### PR TITLE
Remove obsolete iproute APT package check

### PR DESCRIPTION
**Pull request by MichaIng**
**By submitting this pull request, I confirm the following:** 
*please fill any appropriate checkboxes, e.g: [X]*

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [x] I have made only one major change in my proposed changes.
- [x] I have commented my proposed changes within the code.
- [x] I have tested my proposed changes, and have included unit tests where possible.
- [x] I am willing to help maintain this change if there are issues with it later.
- [x] I give this submission freely and claim no ownership.
- [x] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [x] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

Please make sure you [Sign Off](https://github.com/pi-hole/pi-hole/wiki/How-to-signoff-your-commits.) all commits. Pi-hole enforces the [DCO](https://github.com/pi-hole/pi-hole/wiki/Contributing-to-the-project).

---
**What does this PR aim to accomplish?:**
The installer contains a check whether the "iproute2" package is available or "iproute", which was required until Debian Wheezy. Since Debian Jessie, the package is named "iproute2", which is true as well for all supported Ubuntu versions, down to Xenial at least:
- https://packages.debian.org/search?keywords=iproute2
- https://packages.ubuntu.com/search?suite=all&keywords=iproute2

Aside of removing obsolete code, this additionally solves an actual error that can happen when the APT lists/cache update failed with DNS resolving and any package, e.g. from a 3rd party repo, contains the "iproute" name as conflict, "replaces", "provides", alias or similar, which is not uncommon as that name existed. In this case, the installer continues, as "apt-get" does not return an error code when failing at DNS resolving, and without "iproute2" installed or in present APT lists "iproute" could then falsely be selected for install.

Additionally the code comment about apt-get calls with minimal output was moved to the actual apt-get install call section, as if refers to issues with package installs that require interactive action and hence output about it to the console. The package cache update as well requires an interactive confirmation when the underlying suite code name changes, e.g. when "buster" becomes "oldstable" and "bullseye" becomes "stable". But that is not what the referred issue was about.

**What documentation changes (if any) are needed to support this PR?:**
None

---
* You must follow the template instructions. Failure to do so will result in your pull request being closed.
* Please respect that Pi-hole is developed by volunteers, who can only reply in their spare time.

### CODE REVIEW

**Comment by MichaIng** [comment]
It seems to be expected that tests fail on v6 branch?

> expected_stdout = info_box + ' Installing blocking page...'
> ...
> installPiholeWeb: command not found

**Comment by PromoFaux** [comment]
Ah. Yes. I haven't got around to doing the tests yet...!

**Comment by PromoFaux** [comment]
Same here ... 😅

File: automated install/basic-install.sh [hunk 1]
```diff
@@ -2346,7 +2334,7 @@ make_temporary_log() {
 copy_to_install_log() {
     # Copy the contents of file descriptor 3 into the install log
     # Since we use color codes such as '\e[1;33m', they should be removed
-    sed 's/\[[0-9;]\{1,5\}m//g' < /proc/$$/fd/3 > "${installLogLoc}"
+    sed 's/\[[0-9;]\{1,5\}m//g' < /proc/$$/fd/3 > "${installLogLoc}"
```
Comment (MichaIng) [inline automated install/basic-install.sh:2337]: I cleaned the commit history but I have no idea how to add the above `` sign, as it is no UTF-8? Any ideas how to solve this?

Comment (PromoFaux) [reply automated install/basic-install.sh:2337]: It's hex `1B` which is for `ESC`, I believe. https://en.wikipedia.org/wiki/Escape_character

image

Not sure how one types it, though. How did you manage to alter the line?

May be easier just to start fresh perhaps?

Comment (PromoFaux) [reply automated install/basic-install.sh:2337]: This is the result of copy-pasting it into a suggestion, which might work - but I think it's safer to start fresh with something that hasn't touched the line

```suggestion
    sed 's/\[[0-9;]\{1,5\}m//g' < /proc/$$/fd/3 > "${installLogLoc}"
```

Comment (MichaIng) [reply automated install/basic-install.sh]: Okay, I re-added the changes now to a freshly downloaded script.

**Review by PromoFaux — APPROVED** [review]
There was me wondering why the tests were failing - I haven't sorted them out yet.

**Comment by MichaIng** [comment]
I was wondering, too. But it cannot be related to the PR. Trivial enough and I tested the installer, which works fine.

**Comment by PromoFaux** [comment]
Yeah, from a manual overview the code is solid. I'm not too worried about the stability of the `development-v6` branch for the moment, hence I haven't bothered sorting the tests out :)

**Pull request merged**